Makefile

```
@@ -124,4 +124,11 @@ link-check:
 	@command -v lychee >/dev/null 2>&1 || { echo "Install lychee: brew install lychee"; exit 1; }
 	lychee --config lychee.toml --root-dir "$(CURDIR)" '**/*.md'
 
-.PHONY: readme lint lintroller gomodcheck build version build-linux build-windows build-macos deps version-linux version-windows version-macos testacc testacc-cover testacc-coverage test-short test-short-cover generate-mocks link-check
+# Run quick tests with race detector and shuffled order.
+# This target is recommended for CI to catch data races and order-dependent failures.
+# Usage: make test-race
+test-race: deps
+	@echo "Running tests with -race -shuffle=on"
+	go test -race -shuffle=on $(TEST) $(TESTARGS) -timeout 10m
+
+.PHONY: readme lint lintroller gomodcheck build version build-linux build-windows build-macos deps version-linux version-windows version-macos testacc testacc-cover testacc-coverage test-short test-short-cover test-race generate-mocks link-check
```

checkmake (0.2.2)

[warning] Missing required phony target "all"

(minphony)